From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anaconda Recipes <build@anaconda.com>
Date: Wed, 10 Sep 2026 12:00:00 -0600
Subject: [PATCH 0044] Register proto_bazel_features after python_init_rules

Wire companion repos required by protobuf-bazel-rules copied into
@com_google_protobuf by protobuf_bazel_repo (0043).
---
 WORKSPACE                          | 12 ++++--------
 third_party/protobuf_bazel_init.bzl | 10 ++++++++++
 2 files changed, 14 insertions(+), 8 deletions(-)
 create mode 100644 third_party/protobuf_bazel_init.bzl

diff --git a/WORKSPACE b/WORKSPACE
index 0000000..0000000 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -64,6 +64,10 @@ load("@xla//third_party/py:python_init_rules.bzl", "python_init_rules")
 
 python_init_rules()
 
+load("@//third_party:protobuf_bazel_init.bzl", "protobuf_bazel_init")
+
+protobuf_bazel_init()
+
 load("@xla//third_party/py:python_init_repositories.bzl", "python_init_repositories")
 
 python_init_repositories(
@@ -99,14 +103,6 @@ load("@//tensorflow:workspace2.bzl", "tf_workspace2")
 
 tf_workspace2()
 
-# Define system_python repository after protobuf is available.
-# Protobuf's python/BUILD.bazel references @@system_python// which needs to exist.
-load("@com_google_protobuf//python/dist:system_python.bzl", "system_python")
-
-system_python(
-    name = "system_python",
-)
-
 load("@//tensorflow:workspace1.bzl", "tf_workspace1")
 
 tf_workspace1()
diff --git a/third_party/protobuf_bazel_init.bzl b/third_party/protobuf_bazel_init.bzl
new file mode 100644
index 0000000..0000000
--- /dev/null
+++ b/third_party/protobuf_bazel_init.bzl
@@ -0,0 +1,10 @@
+"""Register companion repos required by protobuf-bazel-rules."""
+
+load("@com_google_protobuf//bazel/private/oss:proto_bazel_features.bzl", "proto_bazel_features")
+load("@com_google_protobuf//python/dist:system_python.bzl", "system_python")
+
+def protobuf_bazel_init():
+    """Call after python_init_rules() once @com_google_protobuf exists."""
+    proto_bazel_features(name = "proto_bazel_features")
+    if not native.existing_rule("system_python"):
+        system_python(name = "system_python")
